### ISSUE TITLE
goals_2D_mlps does not have gradients

### ISSUE THREAD

**Issue by FutureOpenAI** [Tsinghua-MARS-Lab/DenseTNT#11]
Thank you for your codes.
I notice that the goals_2D_mlps does not have gradients even for the training. 
Is it expected? And why?

https://github.com/Tsinghua-MARS-Lab/DenseTNT/blob/main/src/modeling/decoder.py#L181-L184

**Comment by GentleSmile** [comment]
No, it is not expected. The operation of detaching is redundant and it does not affect the performance.
